### ISSUE TITLE
Paysafecard

### ISSUE THREAD

**Issue by T0astBread** [liberapay/liberapay.com#1877]
Opening this issue to suggest [paysafecard](https://paysafecard.com/) as an additional payment method. Paysafecard is one of the largest prepaid virtual payment providers in Europe and I think it would be a great addition to Liberapay!

## Why?

From [Wikipedia](https://en.wikipedia.org/wiki/Paysafecard):

> Paysafecard is a prepaid online payment method based on vouchers with a 16-digit PIN code, independent of bank account, credit card, or other personal information. Customers can purchase vouchers at local sales outlets and pay online by entering the code at the checkout of the respective website (e. g. an online game).

The ubiquity of PSC outlets in Europe and the fact that you don't have to provide any personal information to obtain a voucher makes PSC a convenient privacy-preserving online payment method which I think is very in line with Liberapay's mission.

## Why not?

The paysafecard model seems to be mostly based on one-shot payments: Merchants (which is what Liberapay would probably be) initiate a payment, redirect the customer to paysafecard's website, the customer enters their voucher and paysafecard redirects the customer back to the merchant. If everything works out the merchants gets another callback from paysafecard and can collect the funds.

Apologies for not being too familiar with Liberapay's payment model but if I'm not mistaken, Liberapay already has something like an account balance and the ability to pay for multiple donations in advance? Perhaps it would be possible to collect funds this way via paysafecard but I'm not sure how it works in detail.

There might also be legal limitations to recurring anonymous payments. I haven't investigated this deeply but see https://en.wikipedia.org/wiki/Paysafecard#Security_issues (especially bullet point number 3):
> Indeed a new money-laundering law was passed in Germany, effective at the start of 2012. It obliged customers of anonymous prepaid methods (like paysafecard) to undergo an identification procedure when surpassing a monthly spending amount of 100 €.

## Further reading

Paysafecard apparently also offers a solution for recurring payments via "my paysafecard" accounts although that does somewhat defeat the primary reason for introducing paysafecard. "my paysafecard" accounts can however receive payouts _from_ merchants, so that might be of interest.

Paysafecard apparently has some documentation for developers and merchants but I wasn't able to read most of it since their website is unbearably slow and their API docs are broken for me. They do however publish [code examples](https://github.com/paysafecard) and [some kind of API documentation](https://github.com/paysafecard/api-documentation) on GitHub.

If there's interest in this, I'm happy to assist with the implementation where I can. I can also try to get in contact with paysafecard (without dropping Liberapay's name) to get more info if that's needed.

**Comment by Changaco** [comment]
We can't add a payment method that isn't supported by Stripe.

Taler (#1309) will be the solution for donors who wish to remain anonymous… if it ever launches.

**Comment by T0astBread** [comment]
Oh I see. Just to clarify: Taler is acceptable because with Taler, payments (i.e. the actual money-moving) are handled by Taler exchanges whereas with paysafecard Liberapay would need to handle the money themselves and that's not wanted. Did I get that right?

**Comment by Changaco** [comment]
Yes. We're not allowed to handle the money ourselves since we don't have the required license for that.
